### PR TITLE
feat: payment method capability status

### PR DESCRIPTION
**Pull request by frosso**
# Changes proposed in this Pull Request:

Helps with https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1874 - splitting into multiple PRs

- Fixes the styling of the "Pending activation" pill on the onboarding view
- Ensures the "Pending activation" pill is displayed when the status of the payment method capability is `pending`
- Moved the `PaymentMethodCheckbox` component from the `client/components` directory to a location that is closer to where it is used (`client/upe-onboarding-wizard/upe-preview-methods-selector/payment-method-checkbox/index.js`)

**PLEASE NOTE**: in the "old" settings, we showed the "Pending activation" information _also_ when the capability was not present (or `inactive`). I'm not sure that's accurate, so now we're displaying "Pending activation" _only_ when the capability status is `pending`.

Before:
Screen Shot 2021-10-05 at 3 46 54 PM

After:
Screen Shot 2021-10-05 at 4 25 28 PM

After - with fees:
Screen Shot 2021-10-05 at 4 31 51 PM

# Testing instructions
- Ensure you have the `_wcstripe_feature_upe_settings` flag enabled
- Go to http://localhost:8082/wp-admin/admin.php?page=wc_stripe-onboarding_wizard
- Go to the second step
- The capability status pill now reflects the contents of the `capability` object in your account
- The same is also true for the main settings page, with UPE enabled: http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe

### CODE REVIEW

File: client/upe-onboarding-wizard/__tests__/onboarding-wizard-test.js [hunk 1]
```diff
@@ -2,6 +2,11 @@ import React from 'react';
 import { screen, render } from '@testing-library/react';
 import OnboardingWizard from '../onboarding-wizard';
 
+jest.mock(
```
Comment (frosso) [inline client/upe-onboarding-wizard/__tests__/onboarding-wizard-test.js:5]: Resolves some issues with the redux store - no need to set it up here.

File: client/upe-onboarding-wizard/upe-preview-methods-selector/__tests__/add-payment-methods-task.test.js [hunk 2]
```diff
@@ -16,6 +16,10 @@ jest.mock( 'wcstripe/data', () => ( {
 	useSettings: jest.fn(),
 	useCurrencies: jest.fn(),
 } ) );
+jest.mock(
```
Comment (frosso) [inline client/upe-onboarding-wizard/upe-preview-methods-selector/__tests__/add-payment-methods-task.test.js:19]: Resolves some issues with the redux store - no need to set it up in this test.

File: client/upe-onboarding-wizard/upe-preview-methods-selector/add-payment-methods-task.js [hunk 3]
```diff
@@ -17,8 +17,7 @@ import {
 	useEnabledPaymentMethodIds,
 	useSettings,
 } from '../../data';
-import PaymentMethodCheckboxes from '../../components/payment-methods-checkboxes';
-import PaymentMethodCheckbox from '../../components/payment-methods-checkboxes/payment-method-checkbox';
+import PaymentMethodCheckbox from './payment-method-checkbox';
```
Comment (frosso) [inline client/upe-onboarding-wizard/upe-preview-methods-selector/add-payment-methods-task.js:20]: Moved from `client/components` to this location, since it's its only usage

File: client/upe-onboarding-wizard/upe-preview-methods-selector/payment-method-checkbox/index.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import './style.scss';
+
+const InfoIcon = styled( Icon )`
+	fill: #949494;
+`;
+
+const InfoWrapper = styled.div`
+	flex-shrink: 0;
+	width: 20px;
+	height: 20px;
```
Comment (frosso) [inline client/upe-onboarding-wizard/upe-preview-methods-selector/payment-method-checkbox/index.js:21]: I needed to set the width and height here, otherwise the SVG wasn't correctly positioned within its container

File: client/upe-onboarding-wizard/upe-preview-methods-selector/payment-method-checkbox/style.scss [hunk 5]
```diff
@@ -27,16 +31,4 @@
...
-		margin-left: auto;
-		margin-right: 3px;
-	}
-
-	&__info {
-		> * {
-			margin: 0;
-			color: $gray-600;
-		}
-	}
```
Comment (frosso) [inline client/upe-onboarding-wizard/upe-preview-methods-selector/payment-method-checkbox/style.scss:41]: No longer needed, since the structure of the elements have changed

File: includes/payment-methods/class-wc-stripe-upe-payment-method.php [hunk 6]
```diff
@@ -327,7 +327,7 @@ public function get_subtext_messages( $stripe_method_status ) {
 			$text            = __( 'Pending activation', 'woocommerce-gateway-stripe' );
 			$tooltip_content = sprintf(
 				/* translators: %1: Payment method name */
-				esc_attr__( '%1$s won\'t be visible to your customers until you provide the required information. Follow the instructions Stripe has sent to your e-mail.', 'woocommerce-gateway-stripe' ),
+				esc_attr__( '%1$s won\'t be visible to your customers until you provide the required information. Follow the instructions Stripe has sent to your e-mail address.', 'woocommerce-gateway-stripe' ),
```
Comment (frosso) [inline includes/payment-methods/class-wc-stripe-upe-payment-method.php:330]: I ensured the messaging is consistent between the "old" and the "new" settings, so that we have fewer strings to translate floating around.

File: client/settings/general-settings-section/payment-method-setup-help.js [hunk 7]
```diff
@@ -1,36 +0,0 @@
-import { __, sprintf } from '@wordpress/i18n';
```
Comment (frosso) [inline client/settings/general-settings-section/payment-method-setup-help.js:1]: Renamed to `payment-method-capability-status-pill` and moved to `wcstripe/components`

**Comment by harriswong** [comment]
👀

File: client/components/payment-method-capability-status-pill/__tests__/payment-method-capability-status-pill.test.js [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+describe( 'PaymentMethodCapabilityStatusPill', () => {
+	beforeEach( () => {
+		useGetCapabilities.mockReturnValue( {} );
+	} );
+
+	it( 'should render the "Pending activation" text', () => {
+		useGetCapabilities.mockReturnValue( {
+			giropay_payments: 'pending',
+			card_payments: 'active',
```
Comment (harriswong) [inline client/components/payment-method-capability-status-pill/__tests__/payment-method-capability-status-pill.test.js:19]: What does `card_payments` do in this test? Is it so that it resembles closer to the actual `useGetCapabilities()` return values?

File: client/components/payment-method-capability-status-pill/__tests__/payment-method-capability-status-pill.test.js [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+					id="giropay"
+					label="giropay"
+				/>
+			</UpeToggleContext.Provider>
+		);
+
+		expect( container.firstChild ).toBeNull();
+	} );
+
+	it( 'should not render when the capability is "active"', () => {
```
Comment (harriswong) [inline client/components/payment-method-capability-status-pill/__tests__/payment-method-capability-status-pill.test.js:52]: > Can be active, inactive, pending, or unrequested. ([Ref](https://stripe.com/docs/api/capabilities/object#capability_object-status))

I don't think we need to test all of them since our check is either `capabilityStatus === 'pending'` or else return `null`. I think testing just `active` is good for the non-pending case. 👍

File: client/components/payment-method-capability-status-pill/__tests__/payment-method-capability-status-pill.test.js (hunk 8 above)
Comment (frosso) [reply client/components/payment-method-capability-status-pill/__tests__/payment-method-capability-status-pill.test.js:19]: Yes, that's correct!
I also want to ensure that the `pending` status is being read from the `giropay_payments` attribute, and not the `card_payments` one.

File: client/components/payment-method-capability-status-pill/__tests__/payment-method-capability-status-pill.test.js (hunk 9 above)
Comment (frosso) [reply client/components/payment-method-capability-status-pill/__tests__/payment-method-capability-status-pill.test.js:52]: I think I see what you mean. My intention was that in case of `active` or `undefined`, we don't show the messaging (in case somebody modifies the component to erroneously show the messaging on `active`).

Are you recommending removing this test case?

File: client/upe-onboarding-wizard/upe-preview-methods-selector/payment-method-checkbox/index.js [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+`;
+
+const PaymentMethodDescription = ( { id } ) => {
+	const description = paymentMethodsMap[ id ]?.description;
+	if ( ! description ) {
+		return null;
```
Comment (harriswong) [inline client/upe-onboarding-wizard/upe-preview-methods-selector/payment-method-checkbox/index.js:27]: 👍

File: client/settings/general-settings-section/payment-methods-list.js [hunk 11]
```diff
@@ -65,7 +65,7 @@ const GeneralSettingsSection = () => {
 						<PaymentMethodCheckbox id={ method } />
 						<PaymentMethodWrapper>
 							<PaymentMethodDescription
-								id={ isUpeEnabled ? method : null }
```
Comment (harriswong) [inline client/settings/general-settings-section/payment-methods-list.js:68]: Why are we removing `isUpeEnabled`? 

Side note, I just realized that there are 2 `PaymentMethodDescription` components with the same name.

File: client/components/payment-method-capability-status-pill/__tests__/payment-method-capability-status-pill.test.js (hunk 9 above)
Comment (harriswong) [reply client/components/payment-method-capability-status-pill/__tests__/payment-method-capability-status-pill.test.js:52]: I was actually thinking of adding to this test so that it tests `active`, `inactive` and `unrequested` instead of just `active`. In case `capabilityStatus === 'pending'` is updated to `capabilityStatus === 'pending' || ...`. Then I thought that may be too much since it already tested both paths.

Comment (harriswong) [reply client/components/payment-method-capability-status-pill/__tests__/payment-method-capability-status-pill.test.js:52]: > Are you recommending removing this test case?

Absolutely not. :D

File: client/settings/general-settings-section/payment-methods-list.js (hunk 11 above)
Comment (frosso) [reply client/settings/general-settings-section/payment-methods-list.js:68]: The `id` was only used in the `PaymentMethodDescription` to be passed to the `PaymentMethodSetupHelp` component when UPE was enabled/disabled.
Now I moved the logic of checking for whether the UPE was enabled/disabled inside the `PaymentMethodSetupHelp` (which is now renamed to `PaymentMethodCapabilityStatusPill`).

I thought that enabling/disabling the display of the pill based on the `id` being provided or now was not too clear, so I opted to move the consumption of the UPE context inside the pill instead 🤷

Comment (frosso) [reply client/settings/general-settings-section/payment-methods-list.js:68]: > Side note, I just realized that there are 2 `PaymentMethodDescription` components with the same name. 

Haha that's interesting!
I can see why - one is for the "expanded" version in the setting, the other one is for the "short" version on the onboarding.

Not only that, but there's also a `PaymentMethodsDescription` 😆

Comment (harriswong) [reply client/settings/general-settings-section/payment-methods-list.js:68]: Ah i see. 👍

**Review by harriswong — APPROVED** [review]
Tested `wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe` and `wp-admin/admin.php?page=wc_stripe-onboarding_wizard`. Confirmed pill shows.  👍
